src/DynamoCoreWpf/UI/GuidedTour/Step.cs

```diff
@@ -238,19 +266,59 @@ private void UpdatePopupLocationInvoke(Popup popUp)
         /// </summary>
         /// <param name="uiAutomationData">UIAutomation info read from a json file</param>
         /// <param name="enableUIAutomation">Enable/Disable the automation action for a specific UIElement</param>
-        private void ExecuteUIAutomationStep(StepUIAutomation uiAutomationData, bool enableUIAutomation)
+        private void ExecuteUIAutomationStep(StepUIAutomation uiAutomationData, bool enableUIAutomation, Guide.GuideFlow currentFlow)
         {
+            //This section will search the UIElement dynamically in the Dynamo VisualTree in which an automation action will be executed
+            UIElement automationUIElement = Guide.FindChild(MainWindow, uiAutomationData.Name);
+            if (automationUIElement != null)
+                uiAutomationData.UIElementAutomation = automationUIElement;
+          
             switch (uiAutomationData.ControlType.ToUpper())
             {
                 case "MENUITEM":
-                    if (uiAutomationData.UIElementAutomation != null)
+                    if (uiAutomationData.UIElementAutomation == null)
+                    {
+                        return;
+                    }
+                    MenuItem menuEntry = uiAutomationData.UIElementAutomation as MenuItem;
+                    if (menuEntry == null) return;
+                    if (uiAutomationData.Action.ToUpper().Equals("OPEN"))
```

Comment:
We probably want to define these actions in a class, similar to some other hardcodes strings below

Reply:
@QilongTang you mean create a static strings fields right?

Reply:
@RobertGlobant20 I think there are different variations, no? Should this be enum or class?

Reply:
@QilongTang I've fixed using an enum also same medicine applied for the StepUIAutomation.ControlType.
See the fixes in the commit: https://github.com/DynamoDS/Dynamo/pull/12202/commits/2f4bac61303148e4745cb26593d5e87bb7fe61c0